### PR TITLE
【1.1.16】Add error code matching

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

Add error code matching

### Related issues/PRs

Related issues: #4924

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_dml.sql [hunk 1]
```diff
@@ -121,3 +121,8 @@ INSERT INTO `linkis_ps_configuration_config_value` (`config_key_id`, `config_val
 );
 
 
+UPDATE linkis_ps_error_code SET error_regex = "User (\\S+) cannot submit applications to queue (\\S+?)(?=\\))" WHERE  error_code =  "21001";
+
+INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('43016','模块 %s 没有属性 %s ，请确认代码引用是否正常','AttributeError: '(\\S+)' object has no attribute '(\\S+)'',0);
+INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('43017','存在参数无效或拼写错误，请确认 %s 参数正确性','KeyError: (\\(.+\\))',0);
+INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('43018','文件地址( %s )错误，请确认路径是否存在','FileNotFoundError.*No such file or directory\\:\\s'(\\S+)'',0);
```
Comment (casionone) [inline linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_dml.sql:128]: 文件未找到，请确认该路径（%s）是否存在

File: linkis-dist/package/db/linkis_dml.sql [hunk 2]
```diff
@@ -413,7 +413,7 @@ INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type)
 INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('13009','您的任务因为引擎退出（退出可能是引擎进程OOM或者主动kill引擎）导致失败','ERROR EC exits unexpectedly and actively kills the task',0);
 
 -- 21 cluster Authority  22 db Authority
-INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('21001','会话创建失败，用户%s不能提交应用到队列：%s，请联系提供队列给您的人员','User (\\S+) cannot submit applications to queue (\\S+)',0);
+INSERT INTO linkis_ps_error_code (error_code,error_desc,error_regex,error_type) VALUES ('21001','会话创建失败，用户%s不能提交应用到队列：%s，请联系提供队列给您的人员','User (\\S+) cannot submit applications to queue (\\S+?)(?=\\))',0);
```
Comment (casionone) [inline linkis-dist/package/db/linkis_dml.sql:416]: 需要调整 不能正常匹配
```
User A cannot submit applications to queue ide.test)'
User A cannot submit applications to queue ide.test
```

File: linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_dml.sql (hunk 1 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/upgrade/1.4.1_schema/mysql/linkis_dml.sql:128]: done

File: linkis-dist/package/db/linkis_dml.sql (hunk 2 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/linkis_dml.sql:416]: done

**Review by casionone — APPROVED** [review]

**Pull request merged**